src/storage/double_write_buffer.c

```diff
@@ -717,81 +719,86 @@ dwb_destroy_wait_queue (DWB_WAIT_QUEUE * wait_queue, pthread_mutex_t * mutex)
 }
 
 /*
- * dwb_adjust_write_buffer_values () - Adjust double write buffer values.
+ * dwb_adjust_value () - Adjust value to power of 2.
  *
  * return   : Error code.
- * p_double_write_buffer_size (in/out) : Double write buffer size.
- * p_num_blocks (in/out): The number of blocks.
+ * min (in) : minimum of value to be adjusted.
+ * max (in) : maximum of value to be adjusted.
+ * p_value (in/out) : value to be adjusted.
  *
- *  Note: The buffer size must be a multiple of 512 K. The number of blocks must be a power of 2.
+ *  Note: The adjusted value is a power of 2 that is greater than the value.
  */
 STATIC_INLINE void
-dwb_adjust_write_buffer_values (unsigned int *p_double_write_buffer_size, unsigned int *p_num_blocks)
+dwb_adjust_value (unsigned int min, unsigned int max, unsigned int *p_value)
 {
-  unsigned int min_size;
-  unsigned int max_size;
-
-  assert (p_double_write_buffer_size != NULL && p_num_blocks != NULL
-	  && *p_double_write_buffer_size > 0 && *p_num_blocks > 0);
-
-  min_size = DWB_MIN_SIZE;
-  max_size = DWB_MAX_SIZE;
+  unsigned int limit;
 
-  if (*p_double_write_buffer_size < min_size)
+  if (*p_value < min)
+    {
+      *p_value = min;
+    }
+  else if (*p_value > max)
     {
-      *p_double_write_buffer_size = min_size;
+      *p_value = max;
     }
-  else if (*p_double_write_buffer_size > min_size)
+  else if (!IS_POWER_OF_2 (*p_value))
     {
-      if (*p_double_write_buffer_size > max_size)
+      limit = min << 1;
+
+      while (limit < *p_value)
 	{
-	  *p_double_write_buffer_size = max_size;
+	  limit = limit << 1;
 	}
-      else
-	{
-	  /* find smallest number multiple of 512 k */
-	  unsigned int limit1 = min_size;
-
-	  while (*p_double_write_buffer_size > limit1)
-	    {
-	      assert (limit1 <= DWB_MAX_SIZE);
-	      if (limit1 == DWB_MAX_SIZE)
-		{
-		  break;
-		}
-	      limit1 = limit1 << 1;
-	    }
 
-	  *p_double_write_buffer_size = limit1;
-	}
+      *p_value = limit;
     }
 
-  min_size = DWB_MIN_BLOCKS;
-  max_size = DWB_MAX_BLOCKS;
+  assert (IS_POWER_OF_2 (*p_value));
+  assert (*p_value >= min && *p_value <= max);
+}
 
-  assert (*p_num_blocks >= min_size);
+/*
+ * dwb_load_buffer_size () - Load the buffer size value of double write buffer.
+ *
+ * return   : true of false.
+ * p_double_write_buffer_size (in/out) : the buffer size of double write buffer.
+ *
+ */
+STATIC_INLINE bool
+dwb_load_buffer_size (unsigned int *p_double_write_buffer_size)
+{
+  assert (IS_POWER_OF_2 (DWB_MIN_SIZE) && IS_POWER_OF_2 (DWB_MAX_SIZE));
```

Comment:
If the reason you need assert() here is that `DWB_MIN_SIZE` and `DWB_MAX_SIZE`are defined outside of this code, then I think you also need to compare the sizes of `DWB_MIN_SIZE` and `DWB_MAX_SIZE` and check if `DWB_MIN_SIZE` is greater than 512K. what do you think?
- DWB_MAX_SIZE >= DWB_MIN_SIZE
- DWB_MIN_SIZE >= 512 * 1024

And how about mentioning these conditions where they are defined?

Reply:
@Yechan0815  
Please, reply to this comment. 

> @joohok
Values are defined by people, so I added a debug assertion to make sure that the values are valid.
adding the above condition such as "DWB_MAX_SIZE >= DWB_MIN_SIZE" makes it seem clearer.
If the structure doesn't change anymore, I will add conditions to dwb_load_buffer_size and dwb_load_block_count!

I understand what you mean, it will be nice! 

> and sorry, I couldn't understand exactly what "mentioning these conditions where they are defined" means.
Could you please explain a little more in detail?

I mean, it will be better if there is some comment where these values are defined. it is just an opinion.
```
#define DWB_MAX_SIZE          .... /* add some comment that this value is required to be power of 2 */
#define DWB_MIN_SIZE           .... /* add some comment that this value is required to be power of 2  and greater than 512K */
```